services/credential-server/src/utils/lmdb.ts

```diff
@@ -0,0 +1,39 @@
+import { open, RootDatabase } from "lmdb";
+
+class Lmdb {
+  private static instance: Lmdb;
+  private db: RootDatabase;
+
+  private constructor() {
+    this.db = open({
+      path: "./data/lmdb",
+    });
+  }
+
+  public static getInstance(): Lmdb {
+    if (!Lmdb.instance) {
+      Lmdb.instance = new Lmdb();
+    }
+    return Lmdb.instance;
+  }
+
+  public async put(key: string, value: any): Promise<void> {
+    try {
+      await this.db.put(key, value);
+    } catch (error) {
+      console.error("Error putting data in LMDB:", error);
+      throw error;
```

Comment:
There's no point catching and re-throwing if you're just logging. It's cleaner to just fully remove the try catch and let it throw, you'll get all the details.